### PR TITLE
[build]: Remove unnecessary files from release tarballs.

### PR DESCRIPTION
**Pull request by RAOF**
There's no need for our release tarballs to contain a `debian/`
directory, or the GitHub CI metadata.

I'm *pretty* sure that GitHub will respect the git attributes when
generating the tarball.

### CODE REVIEW

**Review by Saviq — CHANGES_REQUESTED** [review]
There's probably more that we could clean up?

```plain
$ unzip -l mir-clean-up-release-tarballs.zip | grep -e debian -e git
      225  2022-05-26 06:40   mir-clean-up-release-tarballs/.gitignore
        0  2022-05-26 06:40   mir-clean-up-release-tarballs/.gitmodules
        0  2022-05-26 06:40   mir-clean-up-release-tarballs/debian/
```

**Comment by Saviq** [comment]
How clean do we want it?

These irk me:
```
bors.toml
spread/**
tools/make_release_tarball
tools/ppa-upload.sh
tools/process_snaps.py
tools/requirements.txt
```

Should they be moved around so that the proposed .gitattributes will cover? Should they be excluded explicitly? Should they stay around?

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.gitattributes export-ignore
+.gitignore     export-ignore
+debian/*       export-ignore
+debian         export-ignore
+.github/*      export-ignore
```
[the remark this answers is not in the document]
Comment (RAOF) [reply .gitattributes]: Hmmmm, yeah. I seem to recall it not working like that before, but maybe I was holding it wrong? 🤷‍♀️

**Review by Saviq — APPROVED** [review]
Yup.

bors merge

**Pull request merged**